### PR TITLE
Flushing store data to temp file in BGREWRITEAOF command and use fsync

### PR DESCRIPTION
**Pull request by ashwaniYDV**
closes https://github.com/DiceDB/dice/issues/397 https://github.com/DiceDB/dice/issues/17

### CODE REVIEW

**Comment by ashwaniYDV** [comment]
@yashs360 @JyotinderSingh please review this PR. This would guarantee that the old AOF file is not corrupted when BGREWRITEAOF happens and would improve durability.

**Comment by ashwaniYDV** [comment]
@arpitbbhayani would you please review

File: core/aof.go [hunk 1]
```diff
@@ -94,19 +95,31 @@ func DumpAllAOF(store *Store) error {
...
 
 	withLocks(func() {
 		for k, obj := range store.store {
 			err = dumpKey(aof, *((*string)(k)), obj)
 		}
 	}, store, WithStoreLock())
 
+	log.Println("flush temp file's data to disk")
+	if err := syscall.Fsync(int(aof.file.Fd())); err != nil {
+		fmt.Println("fsync failed")
```
Comment (lucifercr07) [inline core/aof.go:113]: Are these `Println` statements required? If yes shall we use `fmt. Errorf()` instead?

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

**Comment by amityahav** [comment]
Please wait with that. I have a pr that makes this even more robust and correct

**Comment by ashwaniYDV** [comment]
closing this PR.
thanks @amityahav

**Pull request closed without merging**